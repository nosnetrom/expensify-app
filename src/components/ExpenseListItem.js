import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({id, description, amount, createdAt}) => (
  <p>
    <span className="ExpenseItem__description">
    <Link to={`/edit/${id}`}>{description}</Link>
    </span>
    <span className="ExpenseItem__amount">{amount}</span>
    <span className="ExpenseItem__createdAt">{createdAt}</span>
  </p>
)

export default ExpenseListItem;