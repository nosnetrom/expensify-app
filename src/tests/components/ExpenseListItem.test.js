import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import ExpenseListItem from '../../components/ExpenseListItem';

test('test a single ExpenseListItem', () => {
  const expense = expenses[3];
  const wrapper = shallow(<ExpenseListItem {...expense} />);
  expect(wrapper).toMatchSnapshot();
});