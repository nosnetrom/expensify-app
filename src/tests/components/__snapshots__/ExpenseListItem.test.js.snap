// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test a single ExpenseListItem 1`] = `
<p>
  <span
    className="ExpenseItem__description"
  >
    <Link
      replace={false}
      to="/edit/d"
    >
      Gastronomy
    </Link>
  </span>
  <span
    className="ExpenseItem__amount"
  >
    10000
  </span>
  <span
    className="ExpenseItem__createdAt"
  >
    691200000
  </span>
</p>
`;
